apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -2803,6 +2947,81 @@ static void mapHcpcs(
             });
   }
 
+  public static CareTeamComponent addCareTeamPerforming(
+      ExplanationOfBenefit eob,
+      ItemComponent eobItem,
+      ClaimCareteamrole role,
+      Optional<String> npiValue,
+      Optional<String> upinValue,
+      Optional<Boolean> includeTaxNumbers,
+      String practionerIdNumber,
+      String taxValue) {
+
+    List<Identifier> identifiers = new ArrayList<Identifier>();
+
+    if (npiValue.isPresent()) {
+      identifiers.add(
+          TransformerUtils.createPractionerIdentifier(IdentifierType.NPI, npiValue.get()));
+    }
+
+    if (upinValue.isPresent()) {
+      identifiers.add(
+          TransformerUtils.createPractionerIdentifier(IdentifierType.UPIN, upinValue.get()));
+    }
+
+    if (includeTaxNumbers.orElse(false)) {
+      identifiers.add(TransformerUtils.createPractionerIdentifier(IdentifierType.TAX, taxValue));
+    }
+
+    return addCareTeamPractitionerForPerforming(
+        eob, eobItem, role, identifiers, practionerIdNumber);
+  }
+
+  public static CareTeamComponent addCareTeamPractitionerForPerforming(
+      ExplanationOfBenefit eob,
+      ItemComponent eobItem,
+      ClaimCareteamrole role,
+      List<Identifier> identifiers,
+      String practitionerIdNumber) {
+    // Try to find a matching pre-existing entry.
+    CareTeamComponent careTeamEntry = eob.addCareTeam();
+    // addItem adds and returns, so we want size() not size() + 1 here
+    careTeamEntry.setSequence(eob.getCareTeam().size());
+
+    String practitionerId = "#practitioner-" + practitionerIdNumber;
+
+    Practitioner practitioner = findOrCreateContainedPractioner(eob, practitionerId);
+    practitioner.setIdentifier(identifiers);
+
+    Reference ref = new Reference(practitioner);
+    ref.setId(practitionerId);
```

Comment:
If you comment this and the practitioner.setId out, does the code still work correctly? I think HAPI will autogenerate an ID here, and you should probably let it.